integration/ledger/snapshot_test.go

```diff
@@ -0,0 +1,1049 @@
+			couchProcess = append(couchProcess, configPeerWithCouchDB(setup, org2peer0))
+
+			By("restarting peers with couchDB")
+			setup.startPeers()
+		})
+
+		AfterEach(func() {
+			for _, proc := range couchProcess {
+				proc.Signal(syscall.SIGTERM)
+				Eventually(proc.Wait(), setup.network.EventuallyTimeout).Should(Receive())
+			}
+		})
+
+		It("generates snapshot and bootstraps from snapshots with statecouchdb", func() {
+			org1peer0 := setup.network.Peer("Org1", "peer0")
+			org2peer0 := setup.network.Peer("Org2", "peer0")
+
+			By("deploying legacy chaincode on peers with statecouchdb")
+			nwo.DeployChaincodeLegacy(setup.network, testchannelID, setup.orderer, legacyChaincode)
+
+			By("initializing chaincode and verifying index created")
+			initMarble(setup.network, testchannelID, setup.orderer, org1peer0, "marbles", "marble_indexed")
+			verifySizeIndexExists(setup.network, "testchannel", setup.orderer, org1peer0, "marbles")
+
+			height := helper.getLedgerHeight(org1peer0)
+			By(fmt.Sprintf("adding marbles at height %d", height))
+			testKey := "marble-0"
+			channelPeers := setup.network.PeersWithChannel(testchannelID)
+			helper.invokeMarblesChaincode(legacyChaincode.Name, org1peer0, "initMarble", "marble-0", "blue", "35", "tom")
+			nwo.WaitUntilEqualLedgerHeight(setup.network, testchannelID, height+1, channelPeers...)
+			helper.invokeMarblesChaincode(legacyChaincode.Name, org1peer0, "initMarble", "marble-1", "red", "100", "jerry")
+			nwo.WaitUntilEqualLedgerHeight(setup.network, testchannelID, height+2, channelPeers...)
+
+			By("getting txid from a block before snapshot is generated")
+			txid := getTxIDFromLastBlock(setup.network, org1peer0)
+
+			// verify snapshot commands with different parameters")
+			blockNum := nwo.GetLedgerHeight(setup.network, org1peer0, testchannelID) - 1
+			verifySnapshotRequestCmds(setup.network, org1peer0, testchannelID, blockNum)
+
+			// test 1: generate snapshots on 2 peers for the same blockNum and verify they are same
+			_, snapshotDir2 := generateAndCompareSnapshots(setup.network, setup.orderer, org1peer0, org2peer0, legacyChaincode, blockNum)
+
+			// test 2: bootstrap a peer in an existing org
+			By("bootstrapping a peer (org2peer1) in existing org from a snapshot")
+			org2peer1, couchProc := bootstrapPeerInExistingOrgFromSnapshot(setup, "Org2", "peer1", testchannelID, snapshotDir2, blockNum, true)
+			couchProcess = append(couchProcess, couchProc)
+
+			By("installing legacyChaincode on org2peer1")
+			nwo.InstallChaincodeLegacy(setup.network, legacyChaincode, org2peer1)
+
+			availableHistoryResult := verifyPeerAfterJoinBySnapshot(helper, org2peer1, testchannelID, &legacyChaincode, "", blockNum, txid, testKey, []*marbleHistoryResult{})
+
+			// TODO:
+			// Index does not exist right after InstallChaincodeLegacy, is this expected?
+			// On the other hand, for a peer in new org, index exists right after InstallChaincodeLegacy
```

Comment:
// No, this is not expected... In fact, I would rather suggest to install the chaincode first and then bootstrap in this case so both the sequences are tested.

Reply:
Figured out why. If `--orderer` parameter is not present when running `peer chaincode invoke`, the chaincode CLI command will first make a `cscc` call to retrieve channel config block to get the orderer endpoint. This caused an error (e.g., `cannot serve block [2]. The ledger is bootstrapped from a snapshot. First available block = [6]`) because the channel config block was not available if a peer was bootstrapped from a snapshot and no channel config update was made. After updating couchdb index test to pass orderer parameter, the problem is resolved. 

Outside of this PR, will need to understand if SDK or discovery service has code to access the channel config block.